driver/src/contracts/stablex_contract.rs

```diff
@@ -59,12 +59,86 @@ impl StableXContract for StableXContractImpl {
 
     fn submit_solution(
         &self,
-        _batch_index: U256,
-        _orders: Vec<Order>,
-        _solution: Solution,
+        batch_index: U256,
+        orders: Vec<Order>,
+        solution: Solution,
     ) -> Result<()> {
-        unimplemented!();
+        let account = self
+            .base
+            .account_with_sufficient_balance()
+            .ok_or("Not enough balance to send Txs")?;
+
+        let (owners, order_ids, volumes, prices, token_ids_for_price) =
+            parse_auction_results(orders, solution);
+
+        self.base
+            .contract
+            .call(
+                "submitSolution",
+                (
+                    batch_index,
+                    owners,
+                    order_ids,
+                    volumes,
+                    prices,
+                    token_ids_for_price,
+                ),
+                account,
+                Options::default(),
+            )
+            .wait()
+            .map_err(DriverError::from)
+            .map(|_| ())
+    }
+}
+
+type ContractRecognizedAuctionResults = (Vec<H160>, Vec<U128>, Vec<U128>, Vec<U128>, Vec<U128>);
+
+fn parse_auction_results(
+    orders: Vec<Order>,
+    solution: Solution,
+) -> ContractRecognizedAuctionResults {
+    assert_eq!(
+        orders.len(),
+        solution.executed_buy_amounts.len(),
+        "Received inconsistent auction result data."
+    );
+    // Representing the solution's price vector more compactly as:
+    // sorted_touched_token_ids, non_zero_prices which are logically bound by index.
+    // Example solution.prices = [3, 0, 1] will be transformed into [0, 2], [3, 1]
+    let mut ordered_token_ids: Vec<U128> = vec![];
+    let mut prices: Vec<U128> = vec![];
+    for (token_id, price) in solution.prices.into_iter().enumerate() {
+        if price > 0 {
+            ordered_token_ids.push(U128::from(token_id));
+            prices.push(U128::from(price.to_be_bytes()));
+        }
+    }
+
+    let mut owners: Vec<H160> = vec![];
+    let mut order_ids: Vec<U128> = vec![];
+    let mut volumes: Vec<U128> = vec![];
+    let zipped_amounts = solution
+        .executed_buy_amounts
+        .into_iter()
+        .zip(solution.executed_sell_amounts.into_iter());
+    for (order_index, (_buy_amount, sell_amount)) in zipped_amounts.enumerate() {
+        if sell_amount > 0 {
+            // order was touched!
+            // Note that above condition is only holds for sell orders.
+            owners.push(orders[order_index].account_id);
+            let order_batch_info = orders[order_index]
+                .batch_information
+                .as_ref()
+                .expect("Batch Information on StableX Order");
```

Comment:
nit: "StableX order needs to have batch_information"

Reply:
Ya this is a weird typo. It used to say "Missing ..."